src/managers/GuildMemberManager.js

```diff
@@ -249,6 +268,21 @@ class GuildMemberManager extends CachedManager {
       _data.channel = undefined;
     }
     _data.roles &&= _data.roles.map(role => (role instanceof Role ? role.id : role));
+
+    if (typeof _data.communicationDisabledUntil !== 'undefined') {
+      if (_data.communicationDisabledUntil === null) {
+        _data.communication_disabled_until = null;
+      } else {
+        let date = new Date(_data.communicationDisabledUntil);
+        if (date.getUTCFullYear() <= 2015) {
+          // Assume seconds
+          date = new Date(date.setTime(Date.now() + date.getTime() * 1000));
```

Comment:
What exactly is this doing?

Reply:
This adds seconds. Is there an issue?

Reply:
Seems a bit messy the way you check if the year is before 2015 and add the seconds if it is, couldn’t this be very prone to errors? I think you’d be better off checking if the number is smaller than 28 days in seconds (the max according to the pr) and add that if so

Reply:
What if the limit is changed? Maybe the limit is dropped to the client's: 1 week. Maybe it's increased 30 days, or 50, or 60... Hardcoding a limit isn't an option I would want to take. Also, what's very prone to errors here?

Reply:
The limit will not increase because the current one is the 32-bit integer limit afaik (and as documented in the PR). If it decreases then the value will still work and you’d like Discord throw that error instead. This code seems unnecessarily complex and doesn’t feel accurate at all to me